src/Indicator.vala

```diff
@@ -338,6 +342,74 @@ public class Sound.Indicator : Wingpanel.Indicator {
         return main_grid;
     }
 
+    /* Handles both SMOOTH and non-SMOOTH events (although is only sent SMOOTH normally).
+     * In order to deliver smooth volume changes it:
+     * * ignores whether "natural scrolling" is set since this does not make sense in this context.
+     * * accumulates very small changes until they become significant.
+     * * ignores rapid changes in direction.
+     * * responds to both horizontal and vertical scrolling.
+     * In the case of diagonal scrolling, it takes the larger of the two directions.
+     */
+    private bool handle_scroll_event (Gdk.EventScroll e, out double dir) {
```

Comment:
I love what this function does, but hate that it's necessary. Shouldn't we move this to Granite and/or open an issue against Gtk? I don't think we want to duplicate this logic in all indicators?

Reply:
I do not think this is universal enough to go into Granite/Gtk.  Maybe parts of it could be useful.
I was worried it might be overkill anyway but I did not find a simpler way of getting the effect I wanted.

Reply:
I would open a discussion on Slack about this possibility rather than hold up this PR.

Reply:
Good point.